code/datums/components/trackerhit.dm

```diff
@@ -0,0 +1,23 @@
+/datum/component/timed_tracking
+
+/datum/component/timed_tracking/Initialize(track_time = 20 SECONDS, image_override, flag_override)
+	. = ..()
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	if(!track_time)
+		return COMPONENT_INCOMPATIBLE
+	
+	var/flags = flag_override ? flag_override : MINIMAP_FLAG_MARINE
+	QDEL_IN(src, track_time)
+	if(image_override)
+		SSminimaps.add_marker(parent, flags, image_override)
+		return
+	if(isxeno(parent))
+		var/mob/living/carbon/xenomorph/xeno = parent
+		SSminimaps.add_marker(xeno, flags, image('icons/UI_icons/map_blips.dmi', null, xeno.xeno_caste.minimap_icon))
+	else
+		SSminimaps.add_marker(parent, flags, image('icons/UI_icons/map_blips.dmi', null, "tracker"))
```

Comment:
This is not tested because atoms with existing markers break here
You also cant make it work with current code because minimaps does nto support multiple seperate flag additions

I was going to refactor to make it possible sometime in the near future but this is just broken asis

Reply:
works fine in the video, tested it with xenos even